libafl/src/executors/inprocess/mod.rs

```diff
@@ -431,58 +431,47 @@ pub fn run_observers_and_save_state<E, EM, OF, Z>(
     state: &mut E::State,
     input: &<E::State as UsesInput>::Input,
     fuzzer: &mut Z,
-    event_mgr: &mut EM,
-    exitkind: ExitKind,
+    manager: &mut EM,
+    exit_kind: ExitKind,
 ) where
     E: HasObservers,
     EM: EventFirer<State = E::State> + EventRestarter<State = E::State>,
     OF: Feedback<E::State>,
     E::State: HasExecutions + HasSolutions + HasCorpus,
-    Z: HasObjective<Objective = OF, State = E::State>,
+    Z: HasObjective<Objective = OF, State = E::State>
+        + HasScheduler<State = E::State>
+        + ExecutionProcessor,
 {
-    let mut observers = executor.observers_mut();
-
-    observers
-        .post_exec_all(state, input, &exitkind)
-        .expect("Observers post_exec_all failed");
-
-    let interesting = fuzzer
-        .objective_mut()
-        .is_interesting(state, event_mgr, input, &*observers, &exitkind)
-        .expect("In run_observers_and_save_state objective failure.");
+    let observers = executor.observers_mut();
+    let scheduler = fuzzer.scheduler_mut();
 
-    if interesting {
-        let executions = *state.executions();
-        let mut new_testcase = Testcase::with_executions(input.clone(), executions);
-        new_testcase.add_metadata(exitkind);
-        new_testcase.set_parent_id_optional(*state.corpus().current());
+    if scheduler.on_evaluation(state, input, &*observers).is_err() {
+        log::info!("Failed to call on_evaluation");
```

Comment:
log the error, `if Err(err) = ...` else this is super unhelpful

Reply:
Also it should be a warning

Reply:
or even a panic?

Reply:
no you can't panic inside a crash handler

Reply:
log:error and then exit?

Reply:
yeah